client/ayon_core/tools/utils/thumbnail_paint_widget.py

```diff
@@ -33,19 +47,136 @@ def __init__(self, parent):
         default_image = get_image("thumbnail.png")
         default_pix = paint_image_with_color(default_image, border_color)
 
+        # Video player components
+        self._media_player = None
+        self._video_widget = None
+        self._play_overlay_visible = True
+        self._is_video_content = False
+        self._current_video_path = None
+        self._is_playing = False
+
         self._border_color = border_color
         self._thumbnail_bg_color = thumbnail_bg_color
         self._default_pix = default_pix
 
         self._cached_pix = None
         self._current_pixes = None
         self._has_pixes = False
+        self._current_thumbnail_paths = None
 
         self._bg_color = QtCore.Qt.transparent
         self._use_checker = True
         self._checker_color_1 = QtGui.QColor(89, 89, 89)
         self._checker_color_2 = QtGui.QColor(188, 187, 187)
 
+        # Enable mouse tracking and set focus policy to receive mouse events
+        self.setMouseTracking(True)
+        self.setFocusPolicy(QtCore.Qt.ClickFocus)
+        # Ensure the widget can receive mouse events
+        self.setAttribute(QtCore.Qt.WA_MouseTracking, True)
+        self.setAttribute(QtCore.Qt.WA_AcceptTouchEvents, False)
+
+        self._setup_video_player()
+
+    def _setup_video_player(self):
+        """Setup video player components."""
+        if not HAS_QTMULTIMEDIA:
+            return
+
+        try:
+            # Qt6 API
+            self._media_player = QtMultimedia.QMediaPlayer(self)
+            self._video_widget = QtMultimediaWidgets.QVideoWidget(self)
+            self._video_widget.hide()
+            self._media_player.setVideoOutput(self._video_widget)
+
+            # Loop video playback - Qt6 uses different signal
+            self._media_player.mediaStatusChanged.connect(
+                self._on_media_status_changed
+            )
+        except (AttributeError, TypeError):
+            # Qt5 fallback
```

Comment:
What exactly is different? And why it crashes? The only difference I see is `QMediaPlayer` initialization (one line) but whole logic is duplicated. If it's because `VideoSurface` should be used then do


```python
if hasattr(QtMultimedia.QMediaPlayer, "VideoSurface")"
    self._media_player = QtMultimedia.QMediaPlayer(
        self, QtMultimedia.QMediaPlayer.VideoSurface
    )
else:
    self._media_player = QtMultimedia.QMediaPlayer(self)